src/transformers/quantizers/quantizer_bnb_4bit.py

```diff
@@ -198,7 +198,8 @@ def create_quantized_param(
             for k, v in state_dict.items():
                 if param_name + "." in k:
                     quantized_stats[k] = v
-                    unexpected_keys.remove(k)
+                    if unexpected_keys is not None:
```

Comment:
Are we guaranteed `k` is in `unexpected_keys` if `unexpected_keys` is not None?

```suggestion
                    if unexpected_keys is not None and k in unexpected_keys:
```